app/scripts/views/mixins/modal-settings-panel-mixin.js

```diff
@@ -45,6 +46,12 @@ define(function (require, exports, module) {
       this.closePanel();
     },
 
+    _closePanelReturnToClients: function () {
+      this.navigate('settings/clients');
+      this.closePanel();
+      $.modal.close();
```

Comment:
Hm I'll double check, I don't think that fires because that one only works when you navigate from existing panels, not modals.